README.md

```diff
@@ -1,2 +1,182 @@
+[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-salesforce.types.svg)](https://github.com/ballerina-platform/module-ballerinax-salesforce.types/commits/main)
+[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-salesforce.types/actions/workflows/build-with-bal-test-graalvm.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-salesforce.types/actions/workflows/build-with-bal-test-graalvm.yml)
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+
+Salesforce is a leading customer relationship management (CRM) platform that helps businesses manage and streamline their sales, service, and marketing operations. The [Ballerina Salesforce Connector](https://central.ballerina.io/ballerinax/salesforce/latest) is a project designed to enhance integration capabilities with Salesforce by providing a seamless connection for Ballerina. Notably, this Ballerina project incorporates record type definitions for the base types of Salesforce objects, offering a comprehensive and adaptable solution for developers working on Salesforce integration projects.
+
+## Setup Guide
+
+To customize this project for your Salesforce account and include your custom SObjects, follow the steps below:
+
+### Step 1: Login to Your Salesforce Developer Account
+
+Begin by logging into your [Salesforce Developer Account](https://developer.salesforce.com/).
+
+### Step 2: Generate Open API Specification for Your SObjects
+
+#### Step 2.1: Initiate OpenAPI Document Generation 
+
+Use the following command to send a POST request to start the OpenAPI document generation process.
+
+```bash
+curl -X POST -H "Content-Type: application/json" -H "Authorization: Bearer YOUR_ACCESS_TOKEN" \
+https://MyDomainName.my.salesforce.com/services/data/vXX.X/async/specifications/oas3 \
+-d '{"resources": ["*"]}'
+```
+Replace YOUR_ACCESS_TOKEN and MyDomainName with your actual access token and Salesforce domain. If successful, you'll receive a response with a URI. Extract the locator ID from the URI.
+
+#### Step 2.2: Retrieve the OpenAPI Document
+
+Send a GET request to fetch the generated OpenAPI document using the following command.
+
+```bash
+curl -X GET -H "Content-Type: application/json" -H "Authorization: Bearer YOUR_ACCESS_TOKEN" \
+https://MyDomainName.my.salesforce.com/services/data/vXX.X/async/specifications/oas3/LOCATOR_ID -o oas.json
+```
+Replace YOUR_ACCESS_TOKEN, MyDomainName, and LOCATOR_ID with your actual values.
+
+### Step 3: Configure Cluster Settings
+
+To prevent Out-of-Memory (OOM) issues, execute the following command:
+
+```bash
+export JAVA_OPTS="$JAVA_OPTS -DmaxYamlCodePoints=99999999"
+```
+
+Generate the Ballerina project for the OpenAPI spec using the Ballerina Open API tool with the following commands.
+
+```bash
+bal openapi -i oas.json --mode client --client-methods resource
+```
```

Comment:
Shall we add details to create a new bal project and naming suggestions for package names?

Reply:
1e9ef6955b03e7c594c73c2478ad7134eef4ab04